### ISSUE TITLE
block validation needs to check that miner is part of storage market

### ISSUE THREAD

**Issue by whyrusleeping** [filecoin-project/specs#315]
The storage market actor keeps track of all the miners that are actually part of the official filecoin storage market. Technically, other miner actors may be created and have sectors committed to them (though I suspect that may fail). Additionally, miners who are slashed for breaking consensus rules are evicted from the storage market.

Miners who are not part of the storage market should not be able to mine blocks. We should add a check that ensures the miner in question is an actual storage market miner. 

Open question: Do we make this check at the lookback (the point back in the chain where we pull the power table from?) or from the current state? I think we might need both, as if we just do the lookback, then a slashed miner would be able to still keep making blocks until the lookback period elapses. cc @sternhenri

**Comment by sternhenri** [comment]
Mmmh it's a good question.

The rule you point out would be: 

1. Check at lookback L (ie power table lookback) what the miner set is.
2. Ensure no miners have been slashed since L, and exclude them from set if they have.

That seems fair. There might be the following question: what if a slashing ends up not being on the "main chain?" i.e. we include it too early by looking ahead of L vs waiting to see whether it makes it to L.

This wouldn't happen for consensus faults, though it could for storage faults. So all this to comfort your proposal @whyrusleeping: we should use both, but only include slashing for consensus faults ahead of time, not for storage faults (only once they've made it to L, which will be tied to finality).

**Comment by pooja** [comment]
@sternhenri Is it okay if we make you DRI for this issue?

**Comment by sternhenri** [comment]
sorry for lag here. on it!

**Comment by sternhenri** [comment]
So my comment above is silly: if you were slashed somewhere between lookback and present (for any reason) and that slash did not make it to the final chain, then nor would the block we are currently validating (as a descendent of the one in which the slashing occurred), so we needn't worry about this. @whyrusleeping's simpler solution works.
